pkg/cmd/start/bootstrap.go

```diff
@@ -0,0 +1,287 @@
+package start
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	kubemetrics "github.com/operator-framework/operator-sdk/pkg/kube-metrics"
+	"github.com/operator-framework/operator-sdk/pkg/leader"
+	"github.com/operator-framework/operator-sdk/pkg/metrics"
+	"github.com/operator-framework/operator-sdk/pkg/restmapper"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/exporter/trace/jaeger"
+	"go.opentelemetry.io/otel/global"
+	"google.golang.org/grpc/codes"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis"
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/controller"
+	"github.com/jaegertracing/jaeger-operator/pkg/tracing"
+	"github.com/jaegertracing/jaeger-operator/pkg/upgrade"
+	"github.com/jaegertracing/jaeger-operator/pkg/version"
+)
+
+func bootstrap(ctx context.Context) manager.Manager {
+	tracing.Bootstrap()
+
+	tracer := global.TraceProvider().GetTracer(v1.BootstrapTracer)
+	ctx, span := tracer.Start(ctx, "bootstrap")
+	defer span.End()
+
+	setLogLevel(ctx)
+
+	namespace := getNamespace(ctx)
+
+	buildIdentity(ctx, namespace)
+
+	if viper.GetBool("tracing-enabled") {
+		buildJaegerExporter(ctx)
+	}
+
+	log.WithFields(log.Fields{
+		"os":              runtime.GOOS,
+		"arch":            runtime.GOARCH,
+		"version":         runtime.Version(),
+		"operator-sdk":    version.Get().OperatorSdk,
+		"jaeger-operator": version.Get().Operator,
+		"identity":        viper.GetString(v1.ConfigIdentity),
+		"jaeger":          version.Get().Jaeger,
+	}).Info("Versions")
+
+	if err := leader.Become(ctx, "jaeger-operator-lock"); err != nil {
+		log.Fatal(err)
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		span.SetStatus(codes.Internal)
+		span.SetAttribute(key.String("error", err.Error()))
```

Comment:
From a quick scan of the otel docs, couldn't find a standard definition for `error` attribute, so unclear whether should be a `boolean` type as in opentracing?

Reply:
Something for @pavolloffay perhaps.